qiskit/transpiler/passes/optimization/consolidate_blocks.py

```diff
@@ -159,11 +172,44 @@ def run(self, dag):
                         dag.remove_op_node(node)
                 else:
                     dag.replace_block_with_op(run, unitary, {qubit: 0}, cycle_check=False)
+
+        dag = self._handle_control_flow_ops(dag)
+
         # Clear collected blocks and runs as they are no longer valid after consolidation
         if "run_list" in self.property_set:
             del self.property_set["run_list"]
         if "block_list" in self.property_set:
             del self.property_set["block_list"]
+
+        return dag
+
+    def _handle_control_flow_ops(self, dag):
+        """
+        This is similar to transpiler/passes/utils/control_flow.py except that the
+        collect blocks is redone for the control flow blocks.
+        """
+        from qiskit.transpiler import PassManager
+
+        pass_manager = PassManager()
+        if "run_list" in self.property_set:
+            pass_manager.append(Collect1qRuns())
+        if "block_list" in self.property_set:
+            pass_manager.append(Collect2qBlocks())
+
+        new_consolidate_blocks = self.__class__(
+            force_consolidate=self.force_consolidate,
+            approximation_degree=self._approximation_degree,
+            target=self.target,
+            decomposer=self.decomposer,
+        )
```

Comment:
I had the same question in #9143  but I'm wondering is there a reason we really need a new copy of the pass here? If we could just use `new_consolidate_blocks = self` here it would simplify the code a bit (and we wouldn't need a new `decomposer` argument.

Reply:
My guess is that this was the safest thing to when testing #9143. Originally it was constructed for each block (or maybe each control flow op). @jakelishman suggested lifting it out of the innermost loop. I followed that here. I guess the intended semantics of this pass (or a pass in general) is that they should be immutable ? I don't see an obvious violation for `ConsolidateBlocks`. I can just enable it and run tests. Might be a good idea to add a test to try to break it.

Reply:
With commit 7a34941 a new copy of the pass is no longer made. The original is reused.